libdnf5/base/goal.cpp

```diff
@@ -71,6 +71,19 @@ void add_obsoletes_to_data(const libdnf5::rpm::PackageQuery & base_query, libdnf
     data |= obsoletes_query;
 }
 
+/// Removes from the query available packages with versions lower than or equal
+/// to the installed ones. Similar to filter_upgrades(), but retains also
+/// installed packages and packages whose names are not installed.
+static void remove_older_versions(libdnf5::rpm::PackageQuery & query) {
+    libdnf5::rpm::PackageQuery installed(query.get_base());
+    installed.filter_installed();
```

Comment:
This is very weak suggestion because it does not change the result.
What about to add additional filter  - `latest()`? Installonly packages will be handled in the next query only once.

Reply:
Added. (although I did not measure impact on performance here).